samza-kafka/src/main/java/org/apache/samza/system/kafka/KafkaSystemConsumer.java

```diff
@@ -60,6 +70,7 @@
   private final Config config;
   private final boolean fetchThresholdBytesEnabled;
   private final KafkaSystemConsumerMetrics metrics;
+  private final KafkaStartpointVisitor kafkaStartpointVisitor;
```

Comment:
It doesn't look like this is getting used anywhere.
Maybe at least add a public API for this class, even if you don't wire it in yet.
Ideally, could you put out a PR which shows this being used? That other PR can be separate from this one, but it would be good to see how it gets used.